File: CarlaChronicles-BackOnTrack/bot2.go
package main

import (
  "strconv"
  "fmt"
  "./mymath"
)

func main() {
  fmt.Println("Diameter: " + strconv.FormatFloat(getDiameter(1.0, 6.0), 'f', -1, 64))
}

func getDiameter(distBeforeTurn, distAfterTurn float64) float64 {
  circumference := distAfterTurn - distBeforeTurn
  diameter := 2 * (circumference/mymath.Tau)
  return diameter
}

File: CarlaChronicles-BackOnTrack/bot2.cpp
#include <iostream>
#include <iomanip>
#include <math.h>
#include "mymath.cpp" // Tau not pi


static float getDiameter(float distBeforeTurn, float distAfterTurn)
{
  float circumference = distAfterTurn - distBeforeTurn;
  float diameter = 2.0f * (circumference / TAU);
  return diameter;
}

int main()
{
  std::cout << std::fixed;
  std::cout << "Diameter: ";
  std::cout << std::setprecision(16) << getDiameter(1.0f, 6.0f) << "\n";
}
